File: backend/src/main/java/com/healthcare/backend/payer/service/PayerService.java
package com.healthcare.backend.payer.service;

import com.healthcare.backend.authorization.entity.AuthorizationRequest;

import java.util.List;

public interface PayerService {

    List<AuthorizationRequest> getPendingRequests();

    AuthorizationRequest approve(Long id);

    AuthorizationRequest reject(Long id);

    AuthorizationRequest needMoreInfo(Long id);

}

File: frontend/src/app/pages/payer/service/payer.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../../environments/environment';
import { AuthorizationRequest } from '../../authorization/models/authorization.model';


@Injectable({
  providedIn: 'root'
})
export class PayerService {

  private http = inject(HttpClient);

  private api = `${environment.apiUrl}/payer`;

  getPending(): Observable<AuthorizationRequest[]> {

    return this.http.get<AuthorizationRequest[]>(
      `${this.api}/pending`
    );

  }

  approve(id: number) {

    return this.http.put(
      `${this.api}/${id}/approve`,
      {}
    );

  }

  reject(id: number) {

    return this.http.put(
      `${this.api}/${id}/reject`,
      {}
    );

  }

  needMoreInfo(id: number) {

    return this.http.put(
      `${this.api}/${id}/need-more-info`,
      {}
    );

  }

}